50 movers curriculum

yingniu opened the pull request.

yingniu committed:
Create 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+- "Learn how to use the `Movers` notebooks"
+---
+keypoints:
+- TODO: a list of technologies are supported by the Moving Platform workflow such as Liquid Robotics Wave glider, Teledyne Webb Research Slocum glider, short deployments (hours) down a line of traps, etc. https://oceantrackingnetwork.org/gliders/
```
OTN supports processing of slocum and wave glider detections, detections from other mobile platforms (ship-board receivers, animal-mounted receivers, etc.), and active tracking

yingniu replied:
Incorporated into https://github.com/ocean-tracking-network/node-manager-training/edit/50_movers_curriculum/_episodes/20_movers_notebooks.md

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+---
+keypoints:
+- TODO: a list of technologies are supported by the Moving Platform workflow such as Liquid Robotics Wave glider, Teledyne Webb Research Slocum glider, short deployments (hours) down a line of traps, etc. https://oceantrackingnetwork.org/gliders/
+- TODO "OTN finishes off detections Issues by running Matching and sensor tag processing"
```
OTN is able to match detections collected by moving platforms as long as geolocation data is provided

yingniu replied:
Incorporated into https://github.com/ocean-tracking-network/node-manager-training/edit/50_movers_curriculum/_episodes/20_movers_notebooks.md

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+- TODO "OTN finishes off detections Issues by running Matching and sensor tag processing"
+---
+
+`mssion metadata`, `telemetry data` and `detection data` should be submitted prior to the Moving platform data loading process. 
```
type: mssion

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+- [ ] - NAME match tags to animals (`detections-4` notebook)
+- [ ] - NAME update detection extract table
+
+metadata: **(put metadata plone link here)**
```
instead of Plone - this shoud says "repository". for each of the rows below too

yingniu replied:
Replaced all occurrences of Plone with `repository`

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+
+# Loading Mission Metadata
+
+Moving platform missing metadata should be reported to the Node in the template provided [here](https://members.oceantrack.org/data/data-collection). 
```
typo: missing metadata should be mission metadata

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+# Loading Mission Metadata
+
+Moving platform missing metadata should be reported to the Node in the template provided [here](https://members.oceantrack.org/data/data-collection). 
+This spreadsheet file will contain one or more missions(rows) of the moving platform: identifiers, instruments used and deployment/recovery times.
```
This spreadsheet file will contain one or more missions (rows) of the moving platform and will include mission/platform identifiers, instruments used, and deployment/recovery times.

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+Moving platform missing metadata should be reported to the Node in the template provided [here](https://members.oceantrack.org/data/data-collection). 
+This spreadsheet file will contain one or more missions(rows) of the moving platform: identifiers, instruments used and deployment/recovery times.
+
+1. Visually check if any missing information, inconsistant or formatting issues in the **essential** columns? Column names and example data are shown as below:
```
1. Visually check the mission metadata sheet for any missing information, inconsistencies or formatting issues in the **essential** columns. Column names and expected data formats are shown as below:
 * platform_id: e.g. `1234567`
 * otn_mission_id: e.g. `1234567-202310031456` (Note: otn_mission_id is an internal unique identifier which can be constructed as `platform_id + deploy_date_time digits`).
 * ins_model_no: e.g. `VMT`
 * ins_serial_no: e.g. `130000`
 * deploy_date_time: e.g. `2023-10-03T14:56:00`, in UTC
 * recover_date_time: e.g. `2023-12-03T12:00:00`, in UTC

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+1. `schema`:  'collectioncode'
+    * please edit to include the relevant project code, in lowercase, between the quotes.
+      
+1. `table_suffix`: e.g. `2024_03` - should be the same as in the `movers - 1 - Load Mission Metadata` notebook.
```
we are loading in movers-1, how can the table name be the same as movers-1? I'm confused by this statement

jdpye replied:
looks like it's just in the same notebook but upstream of the second step.

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+2. Run through the `movers - 1 - Load Mission Metadata` notebook to load the spreadsheet into the `mission_table`:
+
+### User Input
+Cell three requires input from you. This information will be used to get the raw detections CSV and to be able to create a new raw table in the database.
```
Cell three requires input from you. This information will be used to load the raw mission metadata and create a new raw table in the database.

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+
+# Loading Telemetry Data
+
+1. Visually check if any missing information, inconsistant or formatting issues in the **essential** columns? Column names and example data are shown as below:
```
**Note**: if you are trying to load multiple missions, the telemetry CSVs for all missions should be joined into one CSV, with the mission_id and platform_id columns completed to differentiate each mission.

Visually check the submitted telemetry CSV for any missing information, inconsistencies or formatting issues in the essential columns. Column names and expected data formats are shown as below:
 * date_time_utc: e.g. `2023-12-13T13:10:12`
 * lat: e.g. `28.33517`
 * lon: e.g. `-80.33734833`.
 * mission_id: e.g. `1234567-202310031456` (**Note**: the value should match the `mission_table`.`otn_mission_id` in the **Loading Mission Metadata** step)
 * platform_id: e.g. `1234567` (**Note**: the value should match the `mission_table`.`platform_id` in the **Loading Mission Metadata** step)

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+ * table_suffix: e.g. `2024_03` (should be the same as in the `movers - 1 - Load Mission Metadata` notebook)
+
+### User Input
+Cell three requires input from you. This information will be used to get the raw detections CSV and to be able to create a new raw table in the database.
```
Cell three requires input from you. This information will be used to get the raw telemtry CSV and to create a new raw table in the database.

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+
+- With the telemetry and mission table, we can now upload the raw detections and promote them to the detections_yyyy_movers tables. 
+- This notebook has the functionalities of `detections - 2 - c_table into detections_yyyy` and `detections - 2b - timedrift calculations` notebooks. The difference is it handles `_movers` tables. 
+1. Run `movers - 3 - Load Detections` notebook till `Load raw dets into detections_yyyy_movers` cell to populate `detections_yyyy_movers` tables and load raw detections into them.
```
typo: till should be "until"

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+1. Run `movers - 3 - Load Detections` notebook till `Load raw dets into detections_yyyy_movers` cell to populate `detections_yyyy_movers` tables and load raw detections into them.
+
+### User Input
+Cell three requires input from you. This information will be used to get the raw detections CSV and to be able to create a new raw table in the database.
```
`This information will be used to get the raw detections CSV and to be able to create a new raw table in the database.` --  I don't think this is what we are doing in this notebook?

yingniu replied:
Changed to mission table.

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+
+`- [ ] - NAME load to detections_yyyy_movers (`movers-2` notebook) **(:fish: put affected years here)**`
+
+- TODO: add troubleshooting tips here: raw detections were note promoted into detections_yyyy_movers.
```
i am not sure what would cause this to happen - sorry!

yingniu replied:
Removed TODO for now.

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+
+`- [ ] - NAME verify timedrift corrections (movers-3 notebook)`
+
+4. Run the nex two cells to detections_yyyy_movers tables. Check off the step in the Gitlab ticket.
```
typo: nex should be "next"

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+# Loading OTN Detections
+
+### User Input
+Cell three requires input from you. This information will be used to get the raw detections CSV and to be able to create a new raw table in the database.
```
`This information will be used to get the raw detections CSV and to be able to create a new raw table in the database.` I don't think this is what we;re doing in this notebook?

CaitlinBate commented on _episodes/20_movers_notebooks.md:
```diff
@@ -0,0 +1,264 @@
+`- [ ] - NAME load formatted telemetry tables (movers-4 notebook) **(:fish: put affected years here)**`
+
+
+2. Run the `create_reduced_telemetry_tables` cell to reduced telemetry tables. Check off the step and record the affected years in the Gitlab ticket.
```
Run the `create_reduced_telemetry_tables` cell to reduce telemetry tables. Check off the step and record the affected years in the Gitlab ticket.

CaitlinBate reviewed (commented):
keypoints:
- TODO: a list of technologies are supported by the Moving Platform workflow such as Liquid Robotics Wave glider, Teledyne Webb Research Slocum glider, short deployments (hours) down a line of traps, etc. https://oceantrackingnetwork.org/gliders/
- TODO "OTN finishes off detections Issues by running Matching and sensor tag processing"

yingniu committed:
Update 20_movers_notebooks.md

Addressing review comments by Shannon

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

yingniu committed:
Update 20_movers_notebooks.md

CaitlinBate commented:
@yingniu my review is above

yingniu committed:
Update 20_movers_notebooks.md

Corrections from @slandovs https://github.com/ocean-tracking-network/node-manager-training/pull/53/files

yingniu committed:
Update 20_movers_notebooks.md

Manually merged corrections by @slandovs https://github.com/ocean-tracking-network/node-manager-training/pull/53/files

yingniu committed:
Update 20_movers_notebooks.md

Manually merged corrections by @slandovs https://github.com/ocean-tracking-network/node-manager-training/pull/53/files

yingniu committed:
Update 20_movers_notebooks.md

Incorporated review comments from Caitlin.

yingniu committed:
Update 20_movers_notebooks.md

Addressed  Caitlin's comments.

yingniu committed:
Update 20_movers_notebooks.md

Addressed  Caitlin's comments.

yingniu committed:
Update 20_movers_notebooks.md

Copy/paste content from 08_Detections.md as @slandovs suggested.

yingniu committed:
Update 20_movers_notebooks.md

Copy/paste 08_Detections.md contents as @slandovs suggested.

CaitlinBate commented:
who's reviews are we sitll waiting for?

jdpye commented:
According to the status, everyone's. I would settle for any two of us before merging though.

CaitlinBate commented:
> According to the status, everyone's. I would settle for any two of us before merging though.

i reviewed already, approved :D  https://github.com/ocean-tracking-network/node-manager-training/pull/51#issuecomment-2045288144

jdpye reviewed (approved):
I think this is comprehensive and good, and I just see a handful of places where we can add more text to guide (especially the new) node managers on what we expect from them w.r.t. values that go into the cells. I think the nicest way to accomplish that is to add this lesson in current form and then invite small revisions to the live-doc as part of a full-curriculum review later on. I am very excited about this workflow being taught to our node managers this winter!
